src/solver/conv_hip_implicit_gemm_bwd_v4r1_xdlops.cpp

```diff
@@ -100,6 +100,8 @@ PerformanceImplicitGemmBwdDataV4R1Xdlops::CalculateGemmABlockCopyPerformancePara
 
         // GemmABlockCopySrcDataPerRead_GemmM also bounded by size of threadwise copy
         SrcDataPerRead_GemmM = gcd(SrcDataPerRead_GemmM, a_data_per_thread_copy);
+        int mMax             = std::max(GemmMPerBlock / BlockSize, 1);
+        SrcDataPerRead_GemmM = gcd(SrcDataPerRead_GemmM, mMax);
```

Comment:
If GemmMPerBlock < BlockSize, then mMax=1, and SrcDataPerRead_GemmM will always be 1. This is not ideal.

Is there a better logic?

Reply:
sometimes it’s 2
So use max=1 may not enough

Reply:
you are right, I will modify it

Reply:
already changed

Reply:
@asleepzzz This doesn't fix the issue. The behaviour will be the same as your previous commit.

I think this is what you are trying to do
```
SrcDataPerRead_GemmM = gcd(SrcDataPerRead_GemmM, a_data_per_thread_copy); // don't change how SrcDataPerRead_GemmM is calculated

const auto a_data_per_thread_copy_gemmm_min = max(GemmMPerBlock / BlockSize, 1); // this is the minimal data a thread need to copy on GemmM dimension, it should no be smaller than 1 or GemmMPerBlock / BlockSize

const auto a_data_per_thread_copy_gemmm = lcm(a_data_per_thread_copy_gemmm_min, SrcDataPerRead_GemmM) // a_data_per_thread_copy_gemmm should be the least common multiple of a_data_per_thread_copy_gemmm_min and SrcDataPerRead_GemmM
```

If these logic matches what you are trying to do, please use them and update the logic for B-block-copy too.

Reply:
Hi Chao,
the logic is in below ,because copy SrcDataPerRead_GemmM to a_data_per_thread_copy_gemmm,
so the results is also correct :
        int mMax             = GemmMPerBlock / BlockSize;
        if(mMax < 1)
            mMax             = 1;
        SrcDataPerRead_GemmM = gcd(SrcDataPerRead_GemmM, mMax);

        // decide threadwise copy lengths
        const auto a_data_per_thread_copy_gemmm = SrcDataPerRead_GemmM;

Reply:
fwd use lcm for gemmKpack due to it's contiguous for gemmkpack, 
I use gcd for gemmM ,I think they are the same

Reply:
> If GemmMPerBlock < BlockSize, then mMax=1, and SrcDataPerRead_GemmM will always be 1. This is not ideal.

However, this issue would still exist, if using your current logic, isn't it?